### PR TITLE
added Injuries.ts Injury.ts bodPart.ts injuryType.ts

### PR DESCRIPTION
**Pull request by syc0n**
generated separate class for Injuries and a single Injury. Also added constants/bodyPart and injuryType

### CODE REVIEW

File: lib/camelot-unchained.d.ts [hunk 1]
```diff
@@ -1184,6 +1183,51 @@ declare module '__camelot-unchained/classes/AbilityComponent' {
...
+    export default Injuries;
+}
+
+declare module '__camelot-unchained/constants/bodyPart' {
+    /**
+      * This Source Code Form is subject to the terms of the Mozilla Public
+      * License, v. 2.0. If a copy of the MPL was not distributed with this
+      * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+      */
+    enum bodyPart {
```
Comment (Shane7) [inline lib/camelot-unchained.d.ts]: The game client should define and export this enum

**Comment by Shane7** [comment]
Is this meant to be a cross component shared functionality or for a certain component?

File: lib/camelot-unchained.d.ts [hunk 2]
```diff
@@ -1184,6 +1183,51 @@ declare module '__camelot-unchained/classes/AbilityComponent' {
...
+        RIGHTLEG = 4,
+        LEFTFEET = 5,
+        RIGHTFEET = 6,
+        LEFTHAND = 7,
+        RIGHTHAND = 8,
+        LEFTARM = 9,
+        RIGHTARM = 10,
+    }
+    export default bodyPart;
+}
```
Comment (Shane7) [inline lib/camelot-unchained.d.ts]: the game client should define/export this enum(they will need it internally) we shouldn't redefine one here and then have to keep them in sync.

File: lib/camelot-unchained.d.ts [hunk 3]
```diff
@@ -1511,6 +1577,20 @@ declare module '__camelot-unchained/constants/baneBoonCategory' {
...
+declare module '__camelot-unchained/constants/injuryType' {
+    /**
+      * This Source Code Form is subject to the terms of the Mozilla Public
+      * License, v. 2.0. If a copy of the MPL was not distributed with this
+      * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+      */
+    enum injuryType {
+        NONE = 0,
+        TYPE1 = 1,
+        TYPE2 = 2,
```
Comment (Shane7) [inline lib/camelot-unchained.d.ts]: I assume these are just place holders? again another enum that should be defined by the game client

File: lib/classes/Character.js [hunk 4]
```diff
@@ -65,4 +65,4 @@ var Character = (function (_Player) {
...
...Jy4vU3BlbGxCb29rJztcbmltcG9ydCBHcm91cCBmcm9tICcuL0dyb3VwJztcbmltcG9ydCBTdGF0cyBmcm9tICcuL1N0YXRzJztcblxuXG5jbGFzcyBDaGFyYWN0ZXIgZXh0ZW5kcyBQbGF5ZXJ7XG5cbiAgaW52ZW50b3J5OiBJbnZlbnRvcnk7XG4gIGJhbmVzOiBCYW5lQm9vbltdO1xuICBib29uczogQmFuZUJvb25bXTtcbiAgZXF1aXBwZWRJdGVtczogRXF1aXBwZWRJdGVtcztcbiAgc3BlbGxCb29rOiBTcGVsbEJvb2s7XG4gIGdyb3VwOiBHcm91cDtcbiAgc3RhdHM6IFN0YXRzO1xuXG5cblxuXG4gIGNvbnN0cnVjdG9yKGNoYXJhY3RlciA9IDxDaGFyYWN0ZXI+e30pIHtcbiAgICBzdXBlcihjaGFyYWN0ZXIpXG4gICAgdGhpcy5pbnZlbnRvcnkgPSBjaGFyYWN0ZXIuaW52ZW50b3J5IHx8IG5ldyBJbnZlbnRvcnkoKTtcbiAgICB0aGlzLmVxdWlwcGVkSXRlbXMgPSBjaGFyYWN0ZXIuZXF1aXBwZWRJdGVtcyB8fCBuZXcgRXF1aXBwZWRJdGVtcygpO1xuICAgIHRoaXMuc3BlbGxCb29rID0gY2hhcmFjdGVyLnNwZWxsQm9vayB8fCBuZXcgU3BlbGxCb29rKCk7XG4gICAgdGhpcy5iYW5lcyA9IGNoYXJhY3Rlci5iYW5lcyB8fCAgPEJhbmVCb29uW10+W107XG4gICAgdGhpcy5iYW5lcyA9IGNoYXJhY3Rlci5ib29ucyB8fCAgPEJhbmVCb29uW10+W107XG4gICAgdGhpcy5ncm91cCA9IGNoYXJhY3Rlci5ncm91cCB8fCAgbnVsbDtcbiAgICB0aGlzLnN0YXRzID0gY2hhcmFjdGVyLnN0YXRzIHx8IG5ldyBTdGF0cygpO1xuICAgIFxuXG5cblxuXG4gIH1cblxuICBzdGF0aWMgY3JlYXRlKCkge1xuICAgIGxldCBhID0gbmV3IENoYXJhY3RlcigpO1xuICAgIHJldHVybiBhO1xuICB9XG5cbn1cblxuZXhwb3J0IGRlZmF1bHQgQ2hhcmFjdGVyO1xuIl0sInNvdXJjZVJvb3QiOiIuLi8ifQ==
```
Comment (Shane7) [inline lib/classes/Character.js]: Should these .js files be checked in?

File: src/ts/classes/Injury.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    this.health = 0;
+    this.maxHealth = 0;
+    this.wounds = 0; 
+  }  
+    
+  setInjury(health: number, maxHealth: number, wounds: number) {
+    this.health = health;
+    this.maxHealth = maxHealth;
+    this.wounds = wounds;        
+  }
```
Comment (Shane7) [inline src/ts/classes/Injury.ts]: The names being used are throwing me off, the class is injury, and it has a type, health, max health and some number of wounds. This doesn't sound like an injury this sounds like a body part? Also why do we need a set and a reset? Couldn't we simply use a set with 0's where ever we called reset? At a min reset should call set with 0's

File: src/ts/classes/Injury.ts [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+  static create() {
+    let a = new Injury();
+    return a;
+  }
+
```
Comment (Shane7) [inline src/ts/classes/Injury.ts]: What is the difference between this and using the class constructor?

File: src/ts/constants/bodyPart.ts [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+  RIGHTLEG = 4,
+  LEFTFEET = 5,
+  RIGHTFEET = 6,
+  LEFTHAND = 7,
+  RIGHTHAND = 8,
+  LEFTARM = 9,
+  RIGHTARM = 10
+};
+
+export default bodyPart;
```
Comment (Shane7) [inline src/ts/constants/bodyPart.ts]: It this placeholder? the game client should define and export this not the UI. Unless you need specific numbers let the compiler decide, makes it easier to add/remove from the list. Spelling, "CHEAST" -> "CHEST", "LEFTFEET" -> "LEFTFOOT", "RIGHTFEET" -> "RIGHTFOOT"

File: src/ts/constants/injuryType.ts [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+enum injuryType {
+  NONE = 0,
+  TYPE1 =  1,
+  TYPE2 = 2
+};
+
+export default injuryType;
```
Comment (Shane7) [inline src/ts/constants/injuryType.ts]: Placeholder I assume? Not sure what these types are for?

File: src/ts/classes/Injury.ts [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+    this.health = injury.health || 0;
+    this.maxHealth = injury.maxHealth || 0;
+    this.wounds = injury.wounds ||  0;
+      
+  }
+  reset(){
+    this.health = 0;
+    this.maxHealth = 0;
+    this.wounds = 0; 
+  }
```
Comment (Shane7) [inline src/ts/classes/Injury.ts]: Why do we need a set and a reset? Couldn't we simply use a set with 0's where ever we called reset? At a min reset should call set with 0's.

File: src/ts/classes/Injuries.ts [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+  }
+    
+  setInjury(part: bodyPart, health: number, maxHealth: number, wounds: number) {
+     this.injury_map[part].setInjury(health, maxHealth, wounds);
+  }
+    
+  static create() {
+    let a = new Injuries();
+    return a;
+  }
```
Comment (Shane7) [inline src/ts/classes/Injuries.ts]: Just use the constructor

File: src/ts/classes/Injury.ts [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+  setInjury(health: number, maxHealth: number, wounds: number) {
+    this.health = health;
+    this.maxHealth = maxHealth;
+    this.wounds = wounds;        
+  }
+
+  static create() {
+    let a = new Injury();
+    return a;
+  }
```
Comment (Shane7) [inline src/ts/classes/Injury.ts]: Use the constructor

File: src/ts/classes/Injuries.ts [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+
```
Comment (Shane7) [inline src/ts/classes/Injuries.ts]: use the class constructor

**Comment by codecorsair** [comment]
This is now out of date, closing.

**Pull request closed without merging**